docs/devices.ipynb

```diff
@@ -0,0 +1,357 @@
+    "op1 = cirq.X(cirq.GridQubit(7 ,7))\n",
+    "\n",
+    "try:\n",
+    "  my_device.validate_operation(op1)\n",
+    "except Exception as e:\n",
+    "  print(e)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "jkgV8lhEB7L5"
+   },
+   "source": [
+    "Above you used a qubit that wasn't on the device and was invalid. Most validate operations also take into account things like supported gates and connectivity as well:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "VM3kDY6QCRPY"
+   },
+   "outputs": [],
+   "source": [
+    "op1 = cirq.H(q1)\n",
+    "op2 = cirq_google.SYC(q1, q3)\n",
+    "\n",
+    "try:\n",
+    "  my_device.validate_operation(op1)\n",
+    "except Exception as e:\n",
+    "  print(e)\n",
+    "\n",
+    "try:\n",
+    "  my_device.validate_operation(op2)\n",
+    "except Exception as e:\n",
+    "  print(e)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "EcTgQg5xCxzG"
+   },
+   "source": [
+    "These validation operations can also be used with full circuits:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "EiolyMKLC262"
+   },
+   "outputs": [],
+   "source": [
+    "my_circuit = cirq.Circuit(\n",
+    "    cirq.PhasedXPowGate(phase_exponent=0.3)(q1),\n",
+    "    cirq.PhasedXPowGate(phase_exponent=0.3)(q2),\n",
+    "    cirq_google.SYC(q1, q2),\n",
+    "    cirq_google.SYC(q2, q3),\n",
+    ")\n",
+    "my_device.validate_circuit(my_circuit)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "-sNjRTCKDb2D"
+   },
+   "source": [
+    "`my_circuit` satisfies all the device constraints and could be run on a Sycamore device."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "BrxB0eCUDvlI"
+   },
+   "source": [
+    "## 2. Metadata features\n",
+    "\n",
+    "Some devices will also expose additional information via the `metadata` property. You can look at the metadata information of the Sycamore device with:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "clsvdcXzEOIV"
+   },
+   "outputs": [],
+   "source": [
+    "metadata = my_device.metadata\n",
+    "metadata"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "glMmeLevElt-"
+   },
+   "source": [
+    "The Sycamore device is a 2d grid device that exposes a `cirq.GridDeviceMetadata` with a uniform set of gates across all the qubits as well as a connectivity graph. You can explore the properties below:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "9Zm_1IsFEiHT"
+   },
+   "outputs": [],
+   "source": [
+    "type(metadata)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "JbwrPD_AETiX"
+   },
+   "outputs": [],
+   "source": [
+    "qubit_set = metadata.qubit_set\n",
```

Comment:
It might be worth explaining what these fields are in bullet points below.

Reply:
FYI: I would still add more about the various attributes in the metadata portion of the class.

Reply:
Done. Added more descriptions and detail.